vcu/Src/drive_by_wire_mock.c

```diff
@@ -440,6 +440,75 @@ static const CLI_Command_Definition_t mcInitCommandDefinition =
     0 /* Number of parameters */
 };
 
+
+BaseType_t setInverterParameter(char *writeBuffer, size_t writeBufferLength, const char *commandString){
+    uint16_t parameterAddress;
+    uint16_t newValue;
+    BaseType_t status;
+    uint16_t readValue = 0;
+    uint32_t currentState = fsmGetState(&fsmHandle); //returns an index
+    BaseType_t paramLen;
+
+    const char *addressParam = FreeRTOS_CLIGetParameter(commandString, 1, &paramLen);
+    const char *valueParam = FreeRTOS_CLIGetParameter(commandString, 2, &paramLen);
+    
+    sscanf(addressParam, "%hu", &parameterAddress);
+    sscanf(valueParam, "%hu", &newValue);
+
+    if (HV_Power_State == HV_Power_State_On){ 
+        COMMAND_OUTPUT("Error: Car is at high voltage. Operation not permitted\n");
+        return pdFALSE;
+    }
+    else if (currentState == STATE_EM_Enable){ 
+        COMMAND_OUTPUT("Error: Car is in drive mode. Operation not permitted\n");
+        return pdFALSE;
+    } else{
+
+
+        status = mcWriteParamCommand(parameterAddress, newValue); 
+        if (status == HAL_OK){
+            COMMAND_OUTPUT("Parameter written successfully. \n");
+        }
+        if (status != HAL_OK){
+            COMMAND_OUTPUT("Failed to write parameter message at address %u\n", parameterAddress);
+            return pdFALSE;
+        }
+
+        HAL_Delay(1000);
+
+        status = mcReadParamCommand(parameterAddress, readValue);
+        if (status == HAL_OK){
+            COMMAND_OUTPUT("Parameter read successfully. \n");
+
+        if (newValue == readValue){
```

Comment:
too many nested if statements, I would avoid nested if statements as much as possible, I don't see it being necessary in this case. too many nested if statements make it harder to follow because you have to remember the previous conditions

Reply:
I would also delete this, check if it's not what you expect, if it's okay then move on, no need to print something